### PR TITLE
test(profile): add cypress tests

### PR DESCRIPTION
**Pull request by divyajindel**
### Proposed behaviour

Refactor Cypress tests for the `Profile` component to use new cypress-component-react framework for testing.
Add `profile.test.js`. 

### Current behaviour

Current process is the regular Cypress approach with BDD.

### Checklist

- [x] Commits follow our style guide
- [ ] Related issues linked in commit messages if required
- [ ] Screenshots are included in the PR if useful
- [ ] All themes are supported if required
- [ ] Unit tests added or updated if required
- [x] Cypress automation tests added or updated if required
- [ ] Storybook added or updated if required
- [ ] Translations added or updated (including creating or amending translation keys table in storybook) if required
- [ ] Typescript `d.ts` file added or updated if required

#### QA

- [ ] Tested in CodeSandbox/storybook
- [ ] Add new Cypress test coverage if required
- [ ] Carbon implementation matches Design System/designs
- [ ] UI Tests GitHub check reviewed if required

### Testing instructions

- [x] Run `npx cypress open --component` to check if there is newly added test.file
- [x] Check if the `profile.test.js` file passed
- [x] Check all tests were moved/refactored from .feature file -> .test.js file
- [x] Check all unnecessary files were deleted, including .json file
- [x] Run `npx cypress run --component` to check none of the other *.test.js files have regressed
- [x] Run `npx cypress run --e2e` to check none of the feature files have regressed

### CODE REVIEW

File: src/components/profile/profile.test.js [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+    });
+
+    it.each([
+      [
+        "https://avataaars.io/?avatarStyle=Transparent&topType=LongHairStraight&accessoriesType=Blank&hairColor=BrownDark&facialHairType=Blank&clotheType=BlazerShirt&eyeType=Default&eyebrowType=Default&mouthType=Default&skinColor=Light",
+      ],
+      [
+        "https://www.gravatar.com/avatar/05c1c705ee45d7ae88b80b3a8866ddaa?s=24&d=404",
+      ],
+    ])("should check avatar for Profile component ", (srcImage) => {
```
Comment (DlgSHi) [inline src/components/profile/profile.test.js:58]:
```suggestion
    ])("should check avatar for Profile component", (srcImage) => {
```

File: src/components/profile/profile.test.js [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+    it.each([
+      [PROFILE_SIZES[0], 22],
+      [PROFILE_SIZES[1], 30],
+      [PROFILE_SIZES[2], 38],
+      [PROFILE_SIZES[3], 54],
+      [PROFILE_SIZES[4], 70],
+      [PROFILE_SIZES[5], 102],
+      [PROFILE_SIZES[6], 126],
+    ])(
+      "should check %s size for Profile component ",
```
Comment (DlgSHi) [inline src/components/profile/profile.test.js:77]:
```suggestion
      "should check %s size for Profile component",
```

File: src/components/profile/profile.test.js [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+          .should("have.css", "height", `${heightAndWidth}px`)
+          .and("have.css", "width", `${heightAndWidth}px`);
+      }
+    );
+
+    it.each([
+      ["Dan Jin", "DJ"],
+      ["Sid Ford", "SF"],
+    ])(
+      "should check initials for %s in Profile component ",
```
Comment (DlgSHi) [inline src/components/profile/profile.test.js:92]:
```suggestion
      "should check initials for %s in Profile component",
```

File: src/components/profile/profile.test.js [hunk 4]
```diff
@@ -0,0 +1,104 @@
+/* eslint-disable no-console */
```
Comment (DlgSHi) [inline src/components/profile/profile.test.js:1]: delete this please

File: src/components/profile/profile.test.js [hunk 5]
```diff
@@ -0,0 +1,104 @@
...
+import * as React from "react";
```
Comment (DlgSHi) [inline src/components/profile/profile.test.js:2]:
```suggestion
import React from "react";
```

File: src/components/profile/profile.test.js [hunk 6]
```diff
@@ -0,0 +1,104 @@
...
+
+import {
+  profilePreview,
+  emailPreview,
+  namePreview,
+  avatarPreview,
+  initialPreview,
+} from "../../../cypress/locators/profile/index";
+
+const testData = ["mp150ú¿¡üßä", "!@#$%^*()_+-=~[];:.,?{}&\"'<>"];
```
Comment (DlgSHi) [inline src/components/profile/profile.test.js:15]: could we import this from `CONSTS` file?

**Review by DipperTheDan — APPROVED** [review]

**Pull request merged**